auditors/nonroot/nonroot.go

```diff
@@ -45,11 +49,65 @@ func (a *RunAsNonRoot) Audit(resource k8stypes.Resource, _ []k8stypes.Resource)
 }
 
 func auditContainer(container *k8stypes.ContainerV1, resource k8stypes.Resource) *kubeaudit.AuditResult {
+	podSpec := k8s.GetPodSpec(resource)
+	if podSpec == nil {
+		return nil
+	}
+
+	if !isContainerRunAsUserNil(container) {
+		if *container.SecurityContext.RunAsUser == 0 {
+			return &kubeaudit.AuditResult{
+				Name:     RunAsUserCSCRoot,
+				Severity: kubeaudit.Error,
+				Message:  "runAsUser is set to UID 0 (root user) in the container SecurityContext. Either set it to a value > 0 or set runAsNonRoot to false.",
```

Comment:
Sorry I missed this the first time 🙈 . Just setting `runAsNonRoot: false` won't actually fix anything if `runAsUser: 0` is set.

```suggestion
				Message:  "runAsUser is set to UID 0 (root user) in the container SecurityContext. Either set it to a value > 0 or remove it and set runAsNonRoot to false.",
```

Reply:
Another nit: `runAsNonRoot` should be  set to `true` so:

`"runAsUser is set to UID 0 (root user) in the container SecurityContext. Either set it to a value > 0 or remove it and set runAsNonRoot to true.",`

Reply:
My bad, didn't saw that too. I've fixed all the messages.